erpnext/stock/doctype/pick_list/pick_list.py

```diff
@@ -34,9 +37,14 @@ def before_submit(self):
 				continue
 			frappe.throw(_('For item {0} at row {1}, count of serial numbers does not match with the picked quantity')
 				.format(frappe.bold(item.item_code), frappe.bold(item.idx)))
-
 		self.set_picked_qty()
 
+	def validate_delivery_date(self):
+		order_delivery_dates = []
+		for location in self.locations:
+			order_delivery_dates.append(frappe.db.get_value("Sales Order Item", location.get("sales_order_item"), "delivery_date"))
+		self.delivery_date = min(order_delivery_dates)		
```

Comment:
Do this conditionally:

```suggestion
		if order_delivery_dates:
			self.delivery_date = min(order_delivery_dates)		
```